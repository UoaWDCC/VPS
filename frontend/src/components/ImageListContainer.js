import React, { useState } from "react";
import ImageList from "@material-ui/core/ImageList";
import ImageListItem from "@material-ui/core/ImageListItem";
import { Box } from "@material-ui/core";

export default function ImageListContainer({ data, onItemSelected }) {
  const rowHeight = 150;
  const [selected, setSelected] = useState();
  const onItemClick = (item) => {
    setSelected(item._id);
    onItemSelected(item);
  };

  return (
    <>
      <ImageList rowHeight={rowHeight} cols={2} gap={10}>
        {data && data.length > 0
          ? data.map((item) => (
              <ImageListItem
                key={item._id}
                cols={1}
                height={rowHeight}
                onClick={() => onItemClick(item)}
              >
                <Box
                  height={rowHeight}
                  border={item._id === selected ? 5 : 0}
                  borderColor={item._id === selected ? "#008a7b" : "#747474"}
                  overflow="hidden"
                  sx={{
                    "&:hover": {
                      opacity: "0.5",
                    },
                    backgroundImage: `url(${item.url})`,
                    backgroundSize: "cover",
                    boxSizing: "border-box",
                  }}
                />
              </ImageListItem>
            ))
          : null}
      </ImageList>
    </>
  );
}
