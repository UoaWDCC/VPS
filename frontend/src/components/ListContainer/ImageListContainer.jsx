import { Box } from "@material-ui/core";
import ImageList from "@material-ui/core/ImageList";
import ImageListItem from "@material-ui/core/ImageListItem";

/**
 * Component used to display images in a list format.
 */
export default function ImageListContainer({ data, onItemSelected, selectedId }) {
  const rowHeight = 150;

  const onItemClick = (item) => {
    onItemSelected(item);
  };

  return (
    <ImageList rowHeight={rowHeight} cols={2} gap={10}>
      {data?.map((item) => (
        <ImageListItem
          key={item.id || item._id} // fallback if some have _id instead
          cols={1}
          onClick={() => onItemClick(item)}
        >
          <Box
            height={rowHeight}
            minWidth={rowHeight}
            border={item.id === selectedId ? 5 : 0}
            borderColor={item.id === selectedId ? "#00b0e6" : "#747474"}
            overflow="hidden"
            sx={{
              "&:hover": {
                opacity: "0.5",
                cursor: "pointer",
              },
              backgroundImage: `url(${item.url})`,
              backgroundSize: "cover",
              backgroundPosition: "center",
              boxSizing: "border-box",
            }}
          />
        </ImageListItem>
      ))}
    </ImageList>
  );
}
