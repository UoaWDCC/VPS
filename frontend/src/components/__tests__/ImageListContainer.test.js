import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import ImageListContainer from "../ImageListContainer";

test("ImageListContainer component snapshot test", () => {
  const testData = [
    {
      _id: "1",
      url: "https://drive.google.com/uc?export=view&id=18XRH_KNKSjhjTKxonEinMSZXnK1OU2at",
    },
    {
      _id: "2",
      url: "https://drive.google.com/uc?export=view&id=1FNZEXQhXc9ieGbnSBBHJ3FfXeMIMai7w",
    },
  ];
  const component = renderer.create(
    <BrowserRouter>
      <ImageListContainer data={testData} />
    </BrowserRouter>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
