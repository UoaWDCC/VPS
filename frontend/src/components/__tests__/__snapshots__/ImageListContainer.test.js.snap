// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageListContainer component snapshot test 1`] = `
<ul
  className="MuiImageList-root"
  style={
    Object {
      "margin": -5,
    }
  }
>
  <li
    className="MuiImageListItem-root"
    height={150}
    onClick={[Function]}
    style={
      Object {
        "height": 160,
        "padding": 5,
        "width": "50%",
      }
    }
  >
    <div
      className="MuiImageListItem-item"
    >
      <div
        className="MuiBox-root MuiBox-root-1"
      />
    </div>
  </li>
  <li
    className="MuiImageListItem-root"
    height={150}
    onClick={[Function]}
    style={
      Object {
        "height": 160,
        "padding": 5,
        "width": "50%",
      }
    }
  >
    <div
      className="MuiImageListItem-item"
    >
      <div
        className="MuiBox-root MuiBox-root-2"
      />
    </div>
  </li>
</ul>
`;
